src/north/CsvToHttp/CsvToHttp.class.js

```diff
@@ -0,0 +1,230 @@
+          if (response) {
+            this.httpBody.push(response)
+          }
+        })
+      }
+    } catch (error) {
+      this.logger.error(error)
+      throw ApiHandler.STATUS.COMMUNICATION_ERROR
+    }
+  }
+
+  /**
+   * @param {Object} jsonObject - jsonObj
+   * @return {Object} - The send status
+   */
+  convertCSVRowIntoHttpBody(jsonObject) {
+    const object = {}
+    this.mapping.forEach((mapping) => {
+      if (jsonObject[mapping.csvField]) {
+        const field = mapping.httpField
+        const value = this.convertToCorrectType(jsonObject[mapping.csvField], mapping.type)
+
+        object[field] = value
+      }
+    })
+    return object
+  }
+
+  /**
+   * Convert the value in the selected type (string by default)
+   * @param {mixed} value - value
+   * @param {string} type - type
+   * @return {mixed} - The converted value
+   */
+  convertToCorrectType(value, type) {
+    let valueConverted
+    switch (type) {
+      case 'integer':
+        try {
+          valueConverted = parseInt(value)
+        } catch (e) {
+          this.logger.error(e)
+          valueConverted = value
+        }
+        break
+      case 'float':
+        try {
+          valueConverted = parseFloat(value)
+        } catch (e) {
+          this.logger.error(e)
+          valueConverted = value
+        }
+        break
+      case 'timestamp':
+        try {
+          valueConverted = parseInt(value)
+        } catch (e) {
+          this.logger.error(e)
+          valueConverted = value
+        }
+        break
+      case 'date (ISO)':
+        try {
+          valueConverted = new Date(value)
+        } catch (e) {
+          this.logger.error(e)
+          valueConverted = value
+        }
+        break
+      default:
+        return value
+    }
+    return valueConverted
+  }
+
+  /**
+   * @return {Promise} - Promise
+   */
+  async sendData() {
+    return new Promise((resolve, reject) => {
+      try {
+        const maxLength = 100
+
+        if (this.httpBody.length > maxLength) {
+          // Divide the current body in array of maximun maxLength elements
+          let i = 0
+          while (i < this.httpBody.length) {
+            this.sendRequest(this.httpBody.slice(i, i + maxLength - 1))
+            i += maxLength
+          }
+        } else {
+          this.sendRequest(this.httpBody)
+        }
+        resolve()
+      } catch (error) {
+        this.logger.error(error)
+        reject()
+      }
+    })
+  }
+
+  /**
+   * Send Request(s) to the selected host
+   * @param {Array} body - Body to send
+   * @return {void}
+   */
+  async sendRequest(body) {
+    try {
+      // Create a base header in order not to send a request with a file but a content
+      const baseHeaders = { 'Content-Type': 'application/json' }
+
+      if (this.request.authentication.type === 'Basic') {
+        this.logger.silly('Sending request with Basic Authentication')
```

Comment:
should also be a switch but in both case, it seems you use the same 
`await this.engine.sendRequest(this.request.host, this.request.method, this.request.authentication, this.proxy, JSON.stringify(body),
          baseHeaders)`

Reply:
Indeed I use the same call to the method (It comes from the change with OibAuthentication)... I will correct it. Should I even use a switch?

Reply:
if it is really the same method, I guess it's better to even not use a switch